src/main/java/org/sqlite/SQLiteJDBCLoader.java

```diff
@@ -420,7 +419,10 @@ public static final class VersionHolder {
                     version = version.trim().replaceAll("[^0-9\\.]", "");
                 }
             } catch (IOException e) {
-                logger.atError()
+                // inline creation of logger to avoid build-time initialization of the logging
+                // framework in native-image
+                LoggerFactory.getLogger(VersionHolder.class)
```

Comment:
It is possible to reference the `SQLiteJDBCLoader#logger` field here.
A caveat of the logger usage here is that if it ever happens in a native-image build, you will end up with the same `Error: Classes that should be initialized at run time got initialized during image building:` instead of the configured log message.

I would argue this is not an issue, as it should never occur in a native-image build. If it happens, normal JVM execution and tests would've already failed anyway.